src/UI.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  EmojiIdentifierResolvable,
+  MessageActionRow,
+  MessageButtonOptions,
+  MessageButtonStyleResolvable,
+} from 'discord.js';
+import { ButtonHandler } from './ButtonHandler';
+import { v4 as uuidv4 } from 'uuid';
+
+export type UIComponent = DispatchButton | DispatchLinkButton;
+
+export type ButtonOptions = {
+  disabled?: boolean;
+  emoji?: EmojiIdentifierResolvable;
+  label?: string;
+  style: Exclude<MessageButtonStyleResolvable, 'LINK'>;
+  onClick: ButtonHandler;
+};
+
+export type LinkButtonOptions = {
```

Comment:
Same with this but use the `Omit` utility to not inherit all properties from `MessageButtonOptions`

Reply:
Same as above